### PR TITLE
Allow user to provide *http.Client when constructing JsonApiGateway

### LINKED ISSUE #4: Allow for custom *http.Client to be passed to the upper clients
Currently, the *http.Client used for making the API calls are hard-coded inside the client different gateway implementations (e.g.: https://github.com/letsencrypt-cpanel/cpanelgo/blob/eb0532054299c19f6aad2f5445e25ea8ff44bbd2/cpanel/authenticated.go#L115)

I think there should be a way to pass an *http.Client that the user might want to use down to the gateway. An easy approach would be to pass the client into the gateway client constructors:

```go
type JsonApiGateway struct {
	Hostname string
	Username string
	Password string
	Insecure bool
	cl       *http.Client
}

func NewJsonApi(hostname, username, password string, insecure bool, cl *http.Client) (CpanelApi, error) {
	c := &JsonApiGateway{
		Hostname: hostname,
		Username: username,
		Password: password,
		Insecure: insecure,
	}

        if cl != nil {
            c.cl = cl
        }

	return CpanelApi{cpanelgo.NewApi(c)}, nil
}
```

What do you think?

### PR DESCRIPTION
**Pull request by drgomesp**
Fixes #4

### CODE REVIEW

**Comment by drgomesp** [comment]
Reopened following @alexzorin's suggestion in #4  .

File: example/cpanelcli.go [hunk 1]
```diff
@@ -130,7 +130,7 @@ func modeCpanel() {
 	required(password, "Please specify a password")
 	required(version, "Please specify an api version")
 
-	cl, err := cpanel.NewJsonApi(hostname, username, password, insecure)
+	cl, err := cpanel.NewJsonApi(hostname, username, password, insecure, nil)
```
Comment (alexzorin) [inline example/cpanelcli.go]: As you have created a separate function, I don't think this change is needed.

**Review by alexzorin — CHANGES_REQUESTED** [review]
Just one small change, otherwise looks good to me.

**Comment by alexzorin** [comment]
Thanks, applied in https://github.com/letsencrypt-cpanel/cpanelgo/commit/0da743896942d3e927a8b735c5497123d89a921d.

**Pull request closed without merging**